var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -223,106 +230,100 @@ def libs(self):
             libraries, root=self.prefix, shared=shared, recursive=True
         )
 
-    @run_before('configure')
     def fortran_check(self):
         if '+fortran' in self.spec and not self.compiler.fc:
             msg = 'cannot build a Fortran variant without a Fortran compiler'
             raise RuntimeError(msg)
 
-    def configure_args(self):
+    def java_check(self):
+        if '+java' in self.spec and not self.compiler.java:
+            msg = 'cannot build a Java variant without a Java compiler'
+            raise RuntimeError(msg)
+
+    def cmake_args(self):
+        args = [
+            '-DBUILD_SHARED_LIBS={0}'.format(
+                'ON' if '+shared' in self.spec else 'OFF')
+        ]
+
         # Always enable this option. This does not actually enable any
         # features: it only *allows* the user to specify certain
         # combinations of other arguments. Enabling it just skips a
         # sanity check in configure, so this doesn't merit a variant.
-        extra_args = ['--enable-unsupported']
-        extra_args += ['--enable-symbols=yes']
-        extra_args += self.enable_or_disable('threadsafe')
-        extra_args += self.enable_or_disable('cxx')
-        extra_args += self.enable_or_disable('hl')
-        extra_args += self.enable_or_disable('fortran')
-        extra_args += self.enable_or_disable('java')
-
-        api = self.spec.variants['api'].value
-        if api != 'none':
-            extra_args.append('--with-default-api-version=' + api)
-
-        if '+szip' in self.spec:
-            szip_spec = self.spec['szip']
-            # The configure script of HDF5 accepts a comma-separated tuple of
-            # two paths: the first one points to the directory with include
-            # files, the second one points to the directory with library files.
-            # If the second path is not specified, the configure script assumes
-            # that it equals to prefix/lib. However, the correct directory
-            # might be prefix/lib64. It is not a problem when the building is
-            # done with Spack's compiler wrapper but it makes the Libtool
-            # files (*.la) invalid, which makes it problematic to use the
-            # installed library outside of Spack environment.
-            extra_args.append('--with-szlib=%s,%s' %
-                              (szip_spec.headers.directories[0],
-                               szip_spec.libs.directories[0]))
+        args.append('-DALLOW_UNSUPPORTED=ON')
+
+        spec = self.spec
+        if '+zlib' in spec:
+            args.append('-DHDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON')
+            args.append(
```

Comment:
Are these arguments necessary?  Spack should be already adding the zlib prefix to the `CMAKE_REFIX_PATH` variable as a dependency so it can be found.

Reply:
Will investigate.

Reply:
Reduced to "args.append(self.define_from_variant('HDF5_ENABLE_Z_LIB_SUPPORT', 'zlib'))"